### PR TITLE
Blackbeard can kill pirates

### PR DESCRIPTION
**Pull request by jyoung4242**

### CODE REVIEW

File: server/utils.ts [hunk 1]
```diff
@@ -50,3 +54,35 @@ export function dist(x1: number, y1: number, x2: number, y2: number) {
...
+      //convert to tiles
+      if (position?.x) {
+        distanceVector.x = Math.abs(position.x - p.x) / 64;
+      }
+      if (position?.y) {
+        distanceVector.y = Math.abs(position.y - p.y) / 64;
+      }
+      const distance = Math.sqrt(distanceVector.x * distanceVector.x + distanceVector.y * distanceVector.y);
+      if (distance <= 3.5) {
+        targetArray.push({ id: p.id, distance: distance });
```
Comment (hpx7) [inline server/utils.ts]: can we use the `dist` function defined above for this?

Comment (jyoung4242) [reply server/utils.ts]: yes

File: client/src/scenes/game.ts [hunk 2]
```diff
@@ -197,11 +238,52 @@ export class GameScene extends Phaser.Scene {
...
+
     state.players.forEach((player) => {
       if (!this.players.has(player.id)) {
         this.addPlayer(player);
       } else {
-        this.updatePlayer(player);
+        if (!player.suspended) {
+          this.updatePlayer(player);
+        } else if (player.suspended && !this.previousSuspendState && player.id === this.user.id) {
+          console.log("here");
```
Comment (hpx7) [inline client/src/scenes/game.ts]: can remove this

File: server/utils.ts [hunk 3]
```diff
@@ -50,3 +54,40 @@ export function dist(x1: number, y1: number, x2: number, y2: number) {
...
+      /*       //Distance = |P-E| = |(3,3)-(1,2)| = |(2,1)| = sqrt(2*2+1*1) = sqrt(5) = 2.23
+      const distanceVector = { x: 0, y: 0 };
+      //convert to tiles
+      if (position?.x) {
+        distanceVector.x = Math.abs(position.x - p.x) / 64;
+      }
+      if (position?.y) {
+        distanceVector.y = Math.abs(position.y - p.y) / 64;
+      }
+      const distance = Math.sqrt(distanceVector.x * distanceVector.x + distanceVector.y * distanceVector.y); */
```
Comment (hpx7) [inline server/utils.ts]: can remove this

File: server/utils.ts [hunk 4]
```diff
@@ -50,3 +54,40 @@ export function dist(x1: number, y1: number, x2: number, y2: number) {
...
+      if (position?.x === undefined || position?.y === undefined) {
+        return undefined;
+      }
+      console.log("positions:", position.x, position.y, p.x, p.y);
```
Comment (hpx7) [inline server/utils.ts]: can remove this

**Pull request merged**